x-pack/legacy/plugins/infra/server/routes/node_details/index.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { boomify } from 'boom';
+import { InfraBackendLibs } from '../../lib/infra_types';
+import { UsageCollector } from '../../usage/usage_collector';
+import { InfraMetricsRequestOptions } from '../../lib/adapters/metrics';
+import { NodeDetailsRequest } from './types';
+import { InfraMetricData } from '../../graphql/types';
+
+export const initNodeDetailsRoute = (libs: InfraBackendLibs) => {
+  const { framework } = libs;
+
+  framework.registerRoute<NodeDetailsRequest, Promise<{ metrics: InfraMetricData[] }>>({
+    method: 'POST',
+    path: '/api/metrics/node_details',
+    handler: async req => {
+      try {
+        const source = await libs.sources.getSourceConfiguration(req, req.payload.sourceId);
```

Comment:
We should create a IO-TS type for the `payload.request` and validate it. Look at the Metadata API to see how that was implemented. That type can also be used in the request hook before the body is converted to JSON.